api/policies/v1/policy_validation.go

```diff
@@ -108,16 +122,22 @@ func validateMatchConditionsField(policy Policy) field.ErrorList {
 		strictCostEnforcement: false,
 	}
 
-	if errs := validateMatchConditions(policy.GetMatchConditions(), opts, field.NewPath("spec").Child("matchConditions")); len(errs) != 0 {
-		return errs
+	return validateMatchConditions(policy.GetMatchConditions(), opts, field.NewPath("spec").Child("matchConditions"))
+}
+
+func validatePolicyServerField(oldPolicy, newPolicy Policy) *field.Error {
+	if oldPolicy.GetPolicyServer() != newPolicy.GetPolicyServer() {
+		return field.Forbidden(field.NewPath("spec").Child("policyServer"), "the field is immutable")
 	}
+
 	return nil
 }
 
-func validatePolicyGroupMembers(policy Policy) *field.Error {
-	if policy.IsPolicyGroup() && len(policy.GetPolicyMembers()) == 0 {
-		return field.Invalid(field.NewPath("spec").Child("policies"), "", "policy groups must have at least one policy member")
+func validatePolicyModeField(oldPolicy, newPolicy Policy) *field.Error {
+	if oldPolicy.GetPolicyMode() != newPolicy.GetPolicyMode() {
```

Comment:
This check is wrong, because it would actually prevent the transition from `monitor` to `protect`.

The old check was this one:

https://github.com/kubewarden/kubewarden-controller/blob/ecfb764b90588a0163e0efa8d86fef22e780c474/api/policies/v1/policy_validation.go#L152-L158

Moreover, if I'm right about this check being wrong, that means our unit tests are not covering the case of a policy changing its mode